sympy/printing/latex.py

```diff
@@ -546,7 +546,7 @@ def convert_args(args):
         # XXX: _print_Pow calls this routine with instances of Pow...
         if isinstance(expr, Mul):
             args = expr.args
-            if args[0] is S.One or any(isinstance(arg, Number) for arg in args[1:]):
+            if args[0] is S.One or any(isinstance(arg, Number) for arg in args[1:]) and args[0].is_number:
```

Comment:
I don't think that `is_number` is the right thing to check here. Why is this needed?

The `is_number` predicate is for checking whether or not an expression can be evaluated using `evalf` to give an explicit approximate complex floating point value. I don't see how that can be the appropriate property to check here.

Reply:
In the current master,
```
>>>latex(Mul(x + y, Rational(1, 2), evaluate=False))
'\\left(x + y\\right) \\frac{1}{2}'
>>> latex(Mul(Rational(1, 2), x + y, evaluate=False))
'\\frac{x + y}{2}'
```
After this change , in this branch,
```
>>>latex(Mul(x + y, Rational(1, 2), evaluate=False))
'\\frac{x + y}{2}'
>>> latex(Mul(Rational(1, 2), x + y, evaluate=False))
'\\frac{x + y}{2}'
```
 But I'm also not sure whether the is_number is plausible or not.

Reply:
These two should not display the same:

> >>>latex(Mul(x + y, Rational(1, 2), evaluate=False))
> '\\frac{x + y}{2}'
> >>> latex(Mul(Rational(1, 2), x + y, evaluate=False))
> '\\frac{x + y}{2}'

It is deliberate that unevaluated expressions should look different from evaluated expressions. In this case the current master output is correct.

Reply:
But here aren't they both unevaluated?

Reply:
> These two should not display the same:
> 
> > > > > latex(Mul(x + y, Rational(1, 2), evaluate=False))
> > > > > '\frac{x + y}{2}'
> > > > > latex(Mul(Rational(1, 2), x + y, evaluate=False))
> > > > > '\frac{x + y}{2}'
> 
> It is deliberate that unevaluated expressions should look different from evaluated expressions. In this case the current master output is correct.
But here aren't they both unevaluated?

Reply:
They are both unevaluated but the first has the Rational as first arg. The `if` check that you are disabling is there deliberately so that expressions with non-canonical args do not print the same as expressions with canonical args.

Reply:
Yes , I understood it now. I'll try some other way of fixing issue #21814

Reply:
@oscarbenjamin I think i found a fix but I've a query:
does `Mul(x - y, S.One, evaluate=False)` should output just `x - y` or `(x - y)*1` , according to my fix it will output the latter is it the correct one?

Reply:
Here `(x - y)*1` is correct.